tests/performance/density-kwok.go

```diff
@@ -117,39 +120,72 @@ var _ = Describe(KWOK("Control Plane Performance Density Testing using kwok", fu
 func createFakeVMIBatchWithKWOK(virtClient kubecli.KubevirtClient, vmCount int) {
 	for i := 1; i <= vmCount; i++ {
 		vmi := newFakeVMISpecWithResources()
+		vmi.Name = fmt.Sprintf("testvmi-%d", i)
 
 		_, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), vmi, metav1.CreateOptions{})
 		Expect(err).ToNot(HaveOccurred())
 	}
 }
 
 func deleteAndVerifyFakeVMIBatch(virtClient kubecli.KubevirtClient, timeout time.Duration) {
-	err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{})
-	Expect(err).NotTo(HaveOccurred())
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
 
-	Eventually(func() []v1.VirtualMachineInstance {
-		vmis, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).List(context.Background(), metav1.ListOptions{})
+	By("Deleting batches of VMIs and verifying all VMIs are deleted")
+	Eventually(func() bool {
+		vmis, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).List(ctx, metav1.ListOptions{Limit: deleteBatchSize})
 		Expect(err).ToNot(HaveOccurred())
 
-		return vmis.Items
-	}, timeout, 10*time.Second).Should(BeEmpty())
+		if len(vmis.Items) == 0 {
+			return true
+		}
+
+		err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).DeleteCollection(
+			ctx,
+			metav1.DeleteOptions{},
+			metav1.ListOptions{Limit: deleteBatchSize},
+		)
+		Expect(err).To(Succeed())
+
+		return false
+	}, timeout, 20*time.Second).Should(BeTrue())
 }
 
 func deleteAndVerifyFakeVMBatch(virtClient kubecli.KubevirtClient, timeout time.Duration) {
-	err := virtClient.VirtualMachine(testsuite.GetTestNamespace(nil)).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{})
-	Expect(err).NotTo(HaveOccurred())
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
 
-	Eventually(func() []v1.VirtualMachineInstance {
-		vmis, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).List(context.Background(), metav1.ListOptions{})
+	By("Deleting batches of VMs")
+	Eventually(func() bool {
+		list, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(nil)).List(ctx, metav1.ListOptions{Limit: 1})
```

Comment:
out of curiosity, is this list call needed? can we remove the list call and ignore the error for when object count in API is 0